ansible/vars/topo_ciscovs-5nodes.yml

```diff
@@ -0,0 +1,211 @@
+topology:
+  host_interfaces:
+    - 0
+    - 1
+    - 2
+    - 3
+    - 4
+    - 5
+    - 6
+    - 7
+    - 8
+    - 9
+    - 10
+    - 11
+    - 12
+    - 13
+    - 14
+    - 15
+    - 16
+    - 17
+    - 18
+    - 19
+    - 20
+    - 21
+    - 22
+    - 23
+    - 24
+    - 25
+    - 26
+    - 27
+  disabled_host_interfaces:
+    - 0
+    - 25
+    - 26
+    - 27
+  VMs:
+    P1:
+      vlans:
+        - 28
+      vm_offset: 0
+    P2:
+      vlans:
+        - 29
+      vm_offset: 1
+    P3:
+      vlans:
+        - 30
+      vm_offset: 2
+    P4:
+      vlans:
+        - 31
+      vm_offset: 3
+  DUT:
+    vlan_configs:
+      default_vlan_config: one_vlan_a
+      one_vlan_a:
+        Vlan1000:
+          id: 1000
+          intfs: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
+          prefix: 192.168.0.1/21
+          prefix_v6: fc02:1000::1/64
+          tag: 1000
+      two_vlan_a:
+        Vlan100:
+          id: 100
+          intfs: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+          prefix: 192.168.0.1/22
+          secondary_subnet: 192.169.0.1/22
+          prefix_v6: fc02:100::1/64
+          tag: 100
+        Vlan200:
+          id: 200
+          intfs: [13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
+          prefix: 192.168.4.1/22
+          prefix_v6: fc02:200::1/64
+          tag: 200
+      four_vlan_a:
+        Vlan1000:
+          id: 1000
+          intfs: [1, 2, 3, 4, 5, 6]
+          prefix: 192.168.0.1/23
+          prefix_v6: fc02:400::1/64
+          tag: 1000
+        Vlan2000:
+          id: 2000
+          intfs: [7, 8, 9, 10, 11, 12]
+          prefix: 192.168.2.1/23
+          prefix_v6: fc02:401::1/64
+          tag: 2000
+        Vlan3000:
+          id: 3000
+          intfs: [13, 14, 15, 16, 17, 18]
+          prefix: 192.168.4.1/23
+          prefix_v6: fc02:402::1/64
+          tag: 3000
+        Vlan4000:
+          id: 4000
+          intfs: [19, 20, 21, 22, 23, 24]
+          prefix: 192.168.6.1/23
+          prefix_v6: fc02:403::1/64
+          tag: 4000
+
+configuration_properties:
+  common:
+    dut_asn: 65100
+    dut_type: ToRRouter
+    swrole: leaf
+    nhipv4: 10.10.246.254
+    nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
+    spine_asn: 65534
+    leaf_asn_start: 64600
+    tor_asn_start: 65500
+    failure_rate: 0
+
+configuration:
+  P1:
+    properties:
+    - common
+    hwsku: cisco-8101-p4-32x100-vs
```

Comment:
From the logical point of view, this topology is exactly the same as the `topo_t0` topology. I don't think it is necessary to define a new topology file.

The difference is that at run time, the testbed is deployed using `cisco-8101-p4-32x100-vs` vsonic image.

In file `vtestbed.yaml`, new testbeds using new DUT `vlab-c-01` are defined. The new DUT `vlab-c-01` is defined to use `hwsku: cisco-8101-p4-32x100-vs`. Combining these information, it would be sufficient to deploy a testbed running `topo_t0` topology and use `hwsku: cisco-8101-p4-32x100-vs` vsonic image for DUT.

For neighbor VMs to use `hwsku: cisco-8101-p4-32x100-vs`, we can define a new neighbor type like `vsonic-cisco`.
The existing neighbor type `vsonic` is to use the legacy `hwsku: Force10-S6100` as neighbor SONiC VM.

Reply:
Different platform's naming conversion is different. For this cisco vsonic platform, it starts from Ethernet0, Ethernet4, ...  different from Force10-S6100's interface name convention, which starts from Ethernet1, Ethernet2, ..

I have some diff as the following to adjust different naming conventions.

*************** configuration:
*** 131,137 ****
        Loopback0:
          ipv4: 100.1.0.29/32
          ipv6: 2064:100::1d/128
!       Ethernet1:
          lacp: 1
        Port-Channel1:
          ipv4: 10.0.0.57/31
--- 132,138 ----
        Loopback0:
          ipv4: 100.1.0.29/32
          ipv6: 2064:100::1d/128
!       Ethernet0:
          lacp: 1
        Port-Channel1:
          ipv4: 10.0.0.57/31

Reply:
Also, in vsonic, we have the following one in vsonic to hardcode hwsku as Force10-S6100. I would like to pass in hwsku to allow we set hwsku properly for each vsonic device.

https://github.com/sonic-net/sonic-mgmt/pull/13785/files#diff-d2d159b58fcbba3419891a4bd372056d2b054e990da4d91631963648af513e25R39

Reply:
I see. Looks good to me.